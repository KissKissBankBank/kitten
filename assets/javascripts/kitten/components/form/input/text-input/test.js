import React from 'react'
import renderer from 'react-test-renderer'
import { TextInput } from './index'

describe('<TextInput />', () => {
  let component

  describe('with `valid` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput valid />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `error` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput error />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `disabled` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput disabled />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `size` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput size="medium" />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `center` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput center />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `name` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput name="foobar" />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `tag` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput tag="input" />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `digits` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput digits={12} />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `textarea` tag', () => {
    beforeEach(() => {
      component = renderer
        .create(<TextInput tag="textarea" name="message" />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `darkBackground` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput darkBackground />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `autoresize` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput tag="autoresize" />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `has=button` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <TextInput has="button" buttonProps={{ children: 'Hello world' }} />,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `has=button buttonIsInset` props', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <TextInput
            has="button"
            buttonIsInset
            buttonProps={{ children: 'Hello world' }}
          />,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `has=icon` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<TextInput has="icon" icon={<svg />} />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `has=icon iconPosition=right` props', () => {
    beforeEach(() => {
      component = renderer
        .create(<TextInput has="icon" icon={<svg />} iconPosition="right" />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `has=limit` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput has="limit" limit={90} />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `has=unit` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInput has="unit" unit="â‚¬" />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `has=unit unitIsWord` props', () => {
    beforeEach(() => {
      component = renderer
        .create(<TextInput has="unit" unit="grand" unitIsWord />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
