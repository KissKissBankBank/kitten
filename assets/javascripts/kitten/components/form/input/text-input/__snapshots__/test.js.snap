// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> with \`autoresize\` prop matches with snapshot 1`] = `
<div
  className="sc-gsDKAQ dTVKjq k-Form-TextInput__textareaContainer"
>
  <textarea
    className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--autoresize k-Form-TextInput--regular"
    disabled={false}
    name="text"
    onChange={[Function]}
    rows={1}
    style={
      Object {
        "lineHeight": "1.125rem",
      }
    }
  />
  <div
    className="k-Form-TextInput__textareaGradient"
  />
</div>
`;

exports[`<TextInput /> with \`center\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--regular k-Form-TextInput--alignCenter"
  disabled={false}
  name="text"
/>
`;

exports[`<TextInput /> with \`darkBackground\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--regular k-Form-TextInput--darkBackground"
  disabled={false}
  name="text"
/>
`;

exports[`<TextInput /> with \`digits\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput-hasDigits k-Form-TextInput-hasDigits_12 k-Form-TextInput--regular"
  disabled={false}
  name="text"
  style={
    Object {
      "--input-content-width": 12,
    }
  }
/>
`;

exports[`<TextInput /> with \`disabled\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--regular k-Form-TextInput--disabled"
  disabled={true}
  name="text"
/>
`;

exports[`<TextInput /> with \`error\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--regular k-Form-TextInput--error"
  disabled={false}
  name="text"
/>
`;

exports[`<TextInput /> with \`name\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--regular"
  disabled={false}
  name="foobar"
/>
`;

exports[`<TextInput /> with \`size\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--regular"
  disabled={false}
  name="text"
/>
`;

exports[`<TextInput /> with \`tag\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--regular"
  disabled={false}
  name="text"
/>
`;

exports[`<TextInput /> with \`textarea\` tag matches with snapshot 1`] = `
<div
  className="sc-gsDKAQ dTVKjq k-Form-TextInput__textareaContainer"
>
  <textarea
    className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--textarea k-Form-TextInput--regular"
    disabled={false}
    name="message"
  />
  <div
    className="k-Form-TextInput__textareaGradient"
  />
</div>
`;

exports[`<TextInput /> with \`valid\` prop matches with snapshot 1`] = `
<input
  className="sc-bdvvtL huvuwE k-Form-TextInput k-Form-TextInput--regular k-Form-TextInput--valid"
  disabled={false}
  name="text"
/>
`;
